### ISSUE TITLE
[Mac OS/docker-machine] Could not connect to rsync container. See build/README.md for setting up remote Docker engine.

### ISSUE THREAD

**Issue by IanLewis** [kubernetes/kubernetes#48586]
I get the following error when trying to build the current master branch on a Mac. From the timestamps you can see that it takes about 25min of "Syncing sources to container" before the error shows up.

```
kubernetes$ make quick-release
+++ [0707 16:15:19] Verifying Prerequisites....
+++ [0707 16:15:20] Building Docker image kube-build:build-bd52ebb501-5-v1.8.3-1
+++ [0707 16:15:23] Syncing sources to container
!!! [0707 16:41:57] Could not connect to rsync container. See build/README.md for setting up remote Docker engine.
!!! [0707 16:41:57] Call tree:
!!! [0707 16:41:57]  1: build/../build/common.sh:433 kube::build::sync_to_container(...)
!!! [0707 16:41:57]  2: build/release.sh:35 kube::build::build_image(...)
make: *** [quick-release] Error 1
```

Docker machine is installed and seems to work well.

```
kubernetes$ docker-machine ls
NAME       ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
default    -        virtualbox   Stopped                                       Unknown       
kube-dev   *        virtualbox   Running   tcp://192.168.99.100:2376           v17.06.0-ce
```

I even upped the memory of the VM to 5GB to no avail. 

After I get the above error I can see that the rsync container is still running, and doesn't look as if it crashed or restarted.

```
kubernetes$ docker ps
CONTAINER ID        IMAGE                                    COMMAND             CREATED             STATUS              PORTS                       NAMES
e66a70a66ee9        kube-build:build-bd52ebb501-5-v1.8.3-1   "/rsyncd.sh"        43 minutes ago      Up 43 minutes       127.0.0.1:32768->8730/tcp   kube-rsync-bd52ebb501-5-v1.8.3-1
```

**Comment by IanLewis** [comment]
@kubernetes/sig-contributor-experience-misc

**Comment by cblecker** [comment]
possibly related to https://github.com/kubernetes/kubernetes/issues/48478

Bug with Docker 17.06.0-ce. Will be fixed in Docker 17.06.1-ce, or you could roll back to Docker 17.03

**Comment by IanLewis** [comment]
I can confirm that downgrading to 17.03 works.

Also of note is that, if using docker-machine, the docker on the VM also needs to be downgraded.
